### PR TITLE
Cluedo game, Christina Persson

### PR DESCRIPTION
**Pull request by Cilpersson**
# What is this
My fifth project during the Technigo bootcamp. 
Assignment: **Build a Cluedo mystery picker**.

## How I started
I did as I have done for every project so far, watching all the weekly material and doing the labs. After that I felt pretty comfortable to get going with the project. 
 
### What I struggled with and should improve for next weeks project
I didn't really struggle with anything project related this week. I was a pretty straight forward task and I enjoyed  it. 

However, I had a goal for myself that if I was done early this week I should not do any deep dives and just allow myself to have a few days off. That was really hard for me and it's something I need to practice on. I did the JavaScript deep dives but I did manage not do dive into the CSS to much. 

### What I'm proud of
I'm proud of the fourth card I've done. It randomises time, am/pm and date. It also formats numbers that are less than 10 to have a zero before the number e.g. 7 is written as 07.  However every year is leap year in my cluedo game so that's something I need to look into how to fix in the future.  

## If I had more time
The site is kind of responsive but I would hav made it better if I had more time, and obviously made it a bit more aesthetically pleasing.

## View it live
Cluedo game website:
https://cluedotime.netlify.com/

### CODE REVIEW

File: code/clue.js [hunk 1]
```diff
@@ -1,59 +1,265 @@
...
+
+const mrsWhite = {
+  firstName: "Skyler",
+  lastName: "White",
+  color: "white",
+  description: "Has at least one coffee stain on her white blouse at all times",
+  age: 45,
+  image: "assets/white.png",
+  occupation: "Accountant"
+};
```
Comment (ericaglimsholt) [inline code/clue.js:59]: You have a lot of information about each person. Maybe that can be outputted in the cards as well?

File: code/clue.js [hunk 2]
```diff
@@ -1,59 +1,265 @@
...
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio"
+];
 
-const weapons = []
+const months = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+const monthsEven = ["04", "06", "09", "11"];
```
Comment (ericaglimsholt) [inline code/clue.js:146]: These are numbers (int) so you don't need the `"` around!

File: code/clue.js [hunk 3]
```diff
@@ -1,59 +1,265 @@
...
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio"
+];
 
-const weapons = []
+const months = [31, 29, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
```
Comment (ericaglimsholt) [inline code/clue.js:145]: What is this? :)

**Review by ericaglimsholt — APPROVED** [review]
Good work on your project!
I loooove the well written PR description! It's so fun to read how it went for you. I like that you have tried something new out and that you managed it to work 👏 I can see that it took some while to get it running but it's awesome that you finished it. If you want to learn more about using dates in javascript I can give you the tip to have a looked at `date()` function! You can read about it [here](https://www.w3schools.com/jsref/jsref_obj_date.asp)

File: code/clue.js (hunk 3 above)
Comment (Cilpersson) [reply code/clue.js:145]: This is my months-list where each number represents the number of days a month. For e.g. months[0] returns 31 which is the number of days in January. With the help of the pre-made randomSelector function it returned a number value. As I wrote in the readme I didn't take non leap years in to account that's why months[1] is 29 insted of 28. I used it both to randomise month and day in the month. 

Not sure if that explanation made sense, but I added the code below and bolded how I used it! Basically I needed the months list to pick both month and also day. This is most def not the prettiest solution but I was fun to make something all by myself.

`const pickDate = () => {
  let year = randomDateSelector(1919, 2019);
  let `**month**` = randomSelector(`**months**`);
  let day = randomDateSelector(1, `**month**`);
  if (month === 29) {
    month = "02";
    day = addZero(day);
  } else if (month % 2 === 0) {
    month = randomSelector(monthsEven);
    day = addZero(day);
  } else {
    month = randomSelector(monthsOdd);
    day = addZero(day);
  }`

Comment (Cilpersson) [reply code/clue.js]: At least I thought I bolded the important data types 🙈

Comment (ericaglimsholt) [reply code/clue.js:145]: Ohhhh, right! I didn't understand that. I though this was the month numbers, you now like month 01 is januari and 12 is december. That is why I was confused. Maybe this variable could be named `daysInMonths` or something that describes it? :)

Comment (Cilpersson) [reply code/clue.js:145]: Haha yes that would def be a better name! It felt super obvious for me but then again I was the one that came up with the idea so it's not really a surprise that I understood what the purpose of it! I'll make sure to think twice about variable names in the future!

**Pull request closed without merging**